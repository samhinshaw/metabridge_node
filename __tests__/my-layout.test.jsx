import React, { Fragment } from 'react';
import { shallow } from 'enzyme';
import Layout from '../components/my-layout';
import mockChildren from '../__mocks__/children';

// Currently running into issues testing this. It does not affect the other test,
// so it most certainly is related to the mock children.

const TestLayout = () => (
  <Fragment>
    <Layout>{mockChildren}</Layout>
  </Fragment>
);

test('Search renders correctly', () => {
  const component = shallow(<TestLayout />);
  expect(component).toMatchSnapshot();
});
