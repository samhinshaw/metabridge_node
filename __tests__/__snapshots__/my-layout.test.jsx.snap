// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TestLayout />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Layout>
        Object {
          "props": Object {
            "id": "welcome-hero",
          },
        }
      </Layout>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Object {
          "props": Object {
            "id": "welcome-hero",
          },
        },
      },
      "ref": null,
      "rendered": Object {
        "props": Object {
          "id": "welcome-hero",
        },
      },
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Layout>
          Object {
            "props": Object {
              "id": "welcome-hero",
            },
          }
        </Layout>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Object {
            "props": Object {
              "id": "welcome-hero",
            },
          },
        },
        "ref": null,
        "rendered": Object {
          "props": Object {
            "id": "welcome-hero",
          },
        },
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
